### PR TITLE
Multi unindex

### PR DESCRIPTION
**Pull request by amirmamo314**

### CODE REVIEW

**Review by gilburd — CHANGES_REQUESTED** [review]

File: src/meeseeks_db/core.clj [hunk 1]
```diff
@@ -107,6 +107,10 @@
   ([conn id delete?]
    (wcar conn (car/get (str "id:" id))) ))
 
+(defn default-multi-id->iid
+  ([conn id delete?]
+   (wcar conn (dorun (map #(car/get (str "id:" %)) id)))))
```
[the remark this answers is not in the document]
Comment (amirmamo314) [reply src/meeseeks_db/core.clj]: changed

File: src/meeseeks_db/core.clj [hunk 2]
```diff
@@ -116,6 +120,12 @@
       (nth db (mod hc (count db))))
     (first db)))
 
+(defn default-multi-id->conn [db id]
```
[the remark this answers is not in the document]
Comment (amirmamo314) [reply src/meeseeks_db/core.clj]: removed

File: src/meeseeks_db/core.clj [hunk 3]
```diff
@@ -200,6 +243,99 @@
...
+            iid->id (doall (flatten (map first (filter (fn [[iid-id id?]] (nil? id?))
+                                             (map list iid-id_list result-id_list)))))
+            iid-index-pairs (filter
+                              (fn [[iid_i index-pairs_i]] (and iid_i (not-empty index-pairs_i)))
+                              (map list iid_list index-pairs_list))
+            iid_list (doall (map first iid-index-pairs))]
+
+            (when (seq iid->id) (wcar conn (apply car/mset iid->id)))
+
+            (doseq [[iid_i index-pairs_i] iid-index-pairs]
```
Comment (gilburd) [inline src/meeseeks_db/core.clj:298]: can't the conn be outside the doseq using 1 connection for all `iid-index-pairs`?

File: src/meeseeks_db/core.clj [hunk 4]
```diff
@@ -105,7 +105,49 @@
                        ))
        iid)))
   ([conn id delete?]
-   (wcar conn (car/get (str "id:" id))) ))
+   (wcar conn (car/get (str "id:" id)))))
+
+(defn default-multi-id->iid
+  ([conn id]
+     (let [iid_keys (doall (map #(str "id:" %) id))
```
[the remark this answers is not in the document]
Comment (amirmamo314) [reply src/meeseeks_db/core.clj]: unnecessary. removed every doall in multi functions and all tests pass successfully

**Review by amirmamo-kontera — DISMISSED** [review]

**Review by amirmamo-kontera — DISMISSED** [review]

**Review by gilburd — APPROVED** [review]

**Pull request merged**